services/apps/alcs/src/alcs/notice-of-intent-decision/notice-of-intent-decision-v2/notice-of-intent-decision-v2.service.ts

```diff
@@ -297,21 +304,71 @@ export class NoticeOfIntentDecisionV2Service {
       components: decisionComponents,
     });
 
-    await this.validateResolutionNumber(
-      createDto.resolutionNumber,
-      createDto.resolutionYear,
-    );
-
     const savedDecision = await this.noticeOfIntentDecisionRepository.save(
       decision,
       {
         transaction: true,
       },
     );
 
+    if (createDto.decisionToCopy) {
+      await this.copyDecisionFields(createDto.decisionToCopy, savedDecision);
+    }
+
     return this.get(savedDecision.uuid);
   }
 
+  private async copyDecisionFields(
+    decisionToCopy: string,
+    decision: NoticeOfIntentDecision,
+  ) {
+    //It is intentional to only copy select fields
+    const existingDecision = await this.get(decisionToCopy);
+    decision.decisionMaker = existingDecision.decisionMaker;
+    decision.outcomeCode = existingDecision.outcomeCode;
+    decision.decisionMakerName = existingDecision.decisionMakerName;
+    decision.isSubjectToConditions = existingDecision.isSubjectToConditions;
+
+    const oldToNewComponentsUuid = new Map<string, string>();
+    const newToOldComponentsUuid = new Map<string, string>();
+    for (const component of existingDecision.components) {
+      const newUuid = v4();
+      oldToNewComponentsUuid.set(component.uuid, newUuid);
+      newToOldComponentsUuid.set(newUuid, component.uuid);
+    }
+
+    decision.components = existingDecision.components.map(
+      (component) =>
+        new NoticeOfIntentDecisionComponent({
+          ...component,
+          uuid: oldToNewComponentsUuid.get(component.uuid),
+          conditions: [],
+        }),
+    );
+    const savedDecision = await this.noticeOfIntentDecisionRepository.save(
+      decision,
+    );
+
+    savedDecision.conditions = existingDecision.conditions.map((condition) => {
+      const conditionsComponents = condition.components?.map(
+        (component) => component.uuid,
+      );
+      return new NoticeOfIntentDecisionCondition({
+        ...condition,
+        uuid: undefined,
+        components: savedDecision.components.filter((component) => {
+          const oldUuid = newToOldComponentsUuid.get(component.uuid);
+          return (
+            !!oldUuid &&
+            conditionsComponents &&
```

Comment:
Nitpick: Do we need `&& conditionsComponents`? Application service doesn't have it

Reply:
This was the slight difference between using ! and using ? on the .components of the decision. I've changed them both to use ? so that they at least match.